src/data/int/intervals.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.int.basic
+import data.nat.intervals
+
+/-!
+# Finite intervals of integers
+
+This file proves that `ℤ` is a `locally_finite_order` and calculates the cardinality of its
+intervals as finsets and fintypes.
+-/
+
+open finset int
+private lemma inj (a : ℤ) : function.injective (λ n : ℕ, (n : ℤ) + a) :=
+λ m n, by { rw [add_left_inj, coe_nat_inj'], exact id }
+
+instance : locally_finite_order ℤ :=
+{ finset_Icc := λ a b, (finset.range (b + 1 - a).to_nat).map
+    ⟨λ n, n + a, inj a⟩,
```

Comment:
Possibly better with `equiv.add_right`.
```suggestion
{ finset_Icc := λ a b, (finset.range (b + 1 - a).to_nat).map
    (equiv.add_right a).to_embedding,
```

Do we have `embedding.add_right` for cancel monoids? That would be slightly slicker.

Reply:
Oh right, there's a coercion happening here that I missed.

Reply:
I think you should make a quick PR adding:
```lean
lemma int.coe_nat_injective : function.injective (coe : ℕ → ℤ) := λ m n, int.coe_nat_inj

/-- `int.has_coe` is an embedding -/
@[simps]
def int.coe_nat_embedding : ℕ ↪ ℤ := ⟨coe, int.coe_nat_injective⟩
```

which makes this line
```suggestion
{ finset_Icc := λ a b, (finset.range (b + 1 - a).to_nat).map
    (int.coe_nat_embedding.trans $ add_left_embedding a),
```

Yes, it's not shorter, but it saves the need for the stray `inj` lemma.